router/xgress_edge_tunnel/factory_wrapper.go

```diff
@@ -172,11 +172,14 @@ func (self *FactoryWrapper) CreateListener(optionsData xgress.OptionsData) (xgre
 }
 
 func (self *FactoryWrapper) CreateDialer(optionsData xgress.OptionsData) (xgress.Dialer, error) {
-	if delegate := self.delegate.Load(); delegate != nil {
-		return delegate.CreateDialer(optionsData)
+	// wait till delegate is created. Once delegate is created, we should also be calling CreateDialer on
+	// the delegate directly, as the factory will get replaced in the registry
+	for {
```

Comment:
do we really want to potentially do this forever or is this another place where a timeout might be in order?

Reply:
I can't think of a case where this wouldn't exit within a reasonable time frame, but added a timeout just to be safe.